main.go

```diff
@@ -110,3 +117,32 @@ func main() {
 	}
 	wg.Wait()
 }
+
+// Create HTTP client, optionally adding a custom CA
+func createHTTPClient(ca string) *http.Client {
+	httpClient := &http.Client{Timeout: 10 * time.Second}
+	if ca != "" {
+		data, err := ioutil.ReadFile(ca)
+		if err != nil {
+			glog.Fatalf("Can't read CA file %s: %v", ca, err)
+		}
+
+		// an error is okay here, we willl only use the custom CA
+		rootCAs, _ := x509.SystemCertPool()
```

Comment:
I'd prefer to save the `err` and check both values. In general, we shouldn't make assumptions about the meaning of other returned values if `err` is ever non-nil.

```go
rootCAs, err := x509.SystemCertPool()
if err != nil || rootCAs == nil {
  rootCAs = x509.NewCertPool()
}
```